main/acle.md

```diff
@@ -1854,6 +1856,142 @@ or compiler-generated functions.
 
 This attribute does not apply to AArch64.
 
+## Function Multi Versioning
+
+The specification for Function Multi Versioning is in `BETA` state and may
+change or be extended in the future.
+
+Function Multi Versioning provides a convenient way to select the most
+appropriate version of a function at runtime. All versions of the function may
+be in the final binary. The compiler generates all supported versions and the
+runtime makes the selection at load time.
+
+The following attributes trigger the multi variant code generation:
+`target_version`, `target_clones`. These attributes can't be mixed. The
+"default" variant means the version of the function that would be generated
+without these attributes.
+
+This feature may be turned off in compile time. In this case the "default"
+variant shall be used.
+
+`__ARM_FEATURE_FUNCTION_MULTI_VERSIONING` is defined to 1 if available.
+
+``` c
+    __attribute__((target_version("name")))
+```
+
+when applied to a function it will become one of the versions. Function with the
+same name may exist with multiple versions in the same translation unit. One
+"default" version of the  function is required to be provided. Implicitly, when
+without attribute, or explicitly providing the "default" in the attribute. If
+only the "default" variant exist it should be linked directly.
```

Comment:
Can we make these an item list? Something like:

```
The attribute `__attribute__((target_version("name")))` expresses the following:

1. when applied to a function it will become one of the versions. Function with the
same name may exist with multiple versions in the same translation unit.
2. One
"default" version of the  function is required to be provided.
3. Implicitly, when
without attribute, or explicitly providing the "default" in the attribute.
4. If
only the "default" variant exist it should be linked directly.
```

Same for the paragraph below about `target_clones`.

In general I prefer to explicitly enumerate/itemize the rules so that they are easier to refer to.

Reply:
ack.